circuit/environment/src/helpers/linear_combination.rs

```diff
@@ -424,6 +424,8 @@ impl<F: PrimeField> Mul<&F> for LinearCombination<F> {
         let mut output = self;
         output.constant *= coefficient;
         output.terms.iter_mut().for_each(|(_, current_coefficient)| *current_coefficient *= coefficient);
+        // If the coefficient of terms are now zero, remove the entries.
```

Comment:
I think you can combine this with the previous line via a filter_map

Reply:
194323a29